contracts/src/Gateway.sol

```diff
@@ -617,4 +621,31 @@ contract Gateway is IGateway, IInitializable {
         assets.assetHubCreateAssetFee = config.assetHubCreateAssetFee;
         assets.assetHubReserveTransferFee = config.assetHubReserveTransferFee;
     }
+
+    /// @inheritdoc IGateway
+    function transact(
+        ParaID destinationChain,
+        OriginKind originKind,
+        uint128 destinationFee,
+        Weight calldata weightAtMost,
+        bytes calldata call
```

Comment:
Do we may need to limit the size of the call data or charge a fee based on the size of call data?

Reply:
I would prefer not enforce that check on chain, could be achieved from UI side, and it's totally up to third party chain.